### PR TITLE
Add the DoctrineAnnotation ruleset to the PHP-CS-Fixer config

### PR DESCRIPTION
**Pull request by meyerbaptiste**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: tests/Fixtures/TestBundle/Entity/CustomActionDummy.php [hunk 1]
```diff
@@ -19,11 +19,11 @@
...
  * @ORM\Entity
  * @ApiResource(itemOperations={
- *     "get"={"method"="GET"},
- *     "custom_normalization"={"route_name"="custom_normalization"}
- * }, collectionOperations={
- *     "get"={"method"="GET"},
- *     "custom_denormalization"={"route_name"="custom_denormalization"}
+ *     "get": {"method": "GET"},
+ *     "custom_normalization": {"route_name": "custom_normalization"}
+ *     }, collectionOperations={
```
Comment (julienfalque) [inline tests/Fixtures/TestBundle/Entity/CustomActionDummy.php]: This line shouldn't be changed. This looks like a bug, I'll open an issue on PHP CS Fixer repository.

File: tests/Fixtures/TestBundle/Entity/RelationEmbedder.php [hunk 2]
```diff
@@ -23,16 +23,16 @@
...
- *     "get"={"method"="GET"},
- *     "put"={"method"="PUT"},
- *     "delete"={"method"="DELETE"},
- *     "custom_get"={"route_name"="relation_embedded.custom_get"},
- *     "custom1"={"path"="/api/custom-call/{id}", "method"="GET"},
- *     "custom2"={"path"="/api/custom-call/{id}", "method"="PUT"},
+ *     "normalization_context": {"groups": {"barcelona"}},
+ *     "denormalization_context": {"groups": {"chicago"}},
+ *     "hydra_context": {"@type": "hydra:Operation", "hydra:title": "A custom operation", "returns": "xmls:string"}
+ *     }, itemOperations={
```
Comment (julienfalque) [inline tests/Fixtures/TestBundle/Entity/RelationEmbedder.php]: Same here.

**Comment by teohhanhui** [comment]
Why the change from `=` to `:`?

**Comment by meyerbaptiste** [comment]
Fixed @teohhanhui 😄

File: tests/Fixtures/TestBundle/Entity/CustomActionDummy.php (hunk 1 above)
Comment (meyerbaptiste) [reply tests/Fixtures/TestBundle/Entity/CustomActionDummy.php]: Indeed, fixed.

**Comment by meyerbaptiste** [comment]
But to answer to your question @teohhanhui, it's because the `DoctrineAnnotations` ruleset uses `:` as default value but the `doctrine_annotation_array_assignment` rule uses `=`.

In the Doctrine documentation they use `:`, in the Symfony documentation they use `=`.

So I can't understand what was the motivation behind the choice of the default value of the rule and the ruleset. And of course, I can't say what is the best choice... Maybe @julienfalque can help.

File: tests/Fixtures/TestBundle/Entity/ThirdLevel.php [hunk 3]
```diff
@@ -40,15 +40,15 @@ class ThirdLevel
     /**
      * @var int
      *
-     * @ORM\Column(type = "integer")
+     * @ORM\Column(type="integer")
```
[the remark this answers is not in the document]
Comment (soyuka) [reply tests/Fixtures/TestBundle/Entity/ThirdLevel.php]: looks so freaking weird with spaces anyway :laughing:

**Comment by julienfalque** [comment]
See https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2992 for some reasoning about the ruleset.

**Comment by teohhanhui** [comment]
I think we should just stick to `=` and with no spaces for both?

**Comment by meyerbaptiste** [comment]
https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/2693 for explanations

**Pull request merged**